framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -52,54 +52,57 @@ private Range(boolean isEmpty) {
         this.to = Long.MIN_VALUE;
     }
 
+    /**
+     * Returns a range with its bounds specified by two parameters, {@code from} and {@code to}. If
+     * {@code from} is greater than {@code to}, returns {@link #NOTHING}.
+     *
+     * @param from the lower bound (inclusive)
+     * @param to the upper bound (inclusive)
+     */
+    private createRangeOrNothing(long from, long to) {
+        if (from <= resultTo) {
+            return new Range(from, to);
+        } else {
+            return NOTHING;
+        }
+    }
+
     @Override
     public String toString() {
         return String.format("[%s..%s]", from, to);
     }
 
-    /** Return true if this range contains everything. */
+    /** Return true if this range contains every {@code long} value. */
     public boolean isEverything() {
-        return this == EVERYTHING;
+        return from == Long.MIN_VALUE && to == Long.MAX_VALUE;
     }
 
-    /** Return true if this range contains nothing. */
+    /** Return true if this range contains no values. */
     public boolean isNothing() {
         return this == NOTHING;
     }
 
     /**
-     * Converts the elements in this range to int type as if they are overflow in the case of
-     * 32-bits int
+     * Creates a 32-bit integral range from this range. Returns INT_EVERYTHING if this range doesn't
+     * fit in 32-bit values.
      */
     public Range intRange() {
-        if (this.isWiderThan(Integer.MAX_VALUE - Integer.MIN_VALUE + 1)) {
+        if (from < Integer.MIN_VALUE || to > Integer.MAX_VALUE) {
             return INT_EVERYTHING;
         } else {
-            int intFrom = (int) this.from;
-            int intTo = (int) this.to;
-            if (intFrom <= intTo) {
-                return new Range(intFrom, intTo);
-            } else {
-                return INT_EVERYTHING;
-            }
+            return this;
         }
```

Comment:
The `Range.intRange()` and `Range.shortRange` is designed to provide precise estimation of the run-time value when casting a integral type with more bits (Long for example) to a integral type with less bits (Integer or Short). For example, if we cast a long type range [2147483648L..2147483649L] to a int type range, the result should be [-2147483648..-2147483647]. However, with the new change, it becomes INT_EVERYTHING.